encoders/video/VideoTorchEncoder/__init__.py

```diff
@@ -0,0 +1,75 @@
+__copyright__ = "Copyright (c) 2020 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+from jina.executors.decorators import batching, as_ndarray
+import numpy as np
+from jina.executors.encoders import BaseVideoEncoder
+from jina.executors.encoders.frameworks import BaseTorchEncoder
+
+
+class VideoTorchEncoder(BaseTorchEncoder, BaseVideoEncoder):
+    """
+    :class:`VideoTorchEncoder` encodes data from a ndarray, potentially B x T x (Channel x Height x Width) into an
+        ndarray of `B x D`.
+    Internally, :class:`VideoTorchEncoder` wraps the models from `torchvision.models`.
+    https://pytorch.org/docs/stable/torchvision/models.html
+    """
+
+    def __init__(self,
+                 model_name: str = 'r3d_18',
+                 channel_axis: int = 1,
+                 pool_strategy: str = 'mean',
+                 *args, **kwargs):
+        """
+        :param model_name: the name of the model. Supported models include ``r3d_18``, ``mc3_18``, ``r2plus1d_18``
+        :param pool_strategy: the pooling strategy
+            - `None` means that the output of the model will be the 4D tensor output of the last convolutional block.
+            - `mean` means that global average pooling will be applied to the output of the last convolutional block, and
+                 thus the output of the model will be a 2D tensor.
+            - `max` means that global max pooling will be applied.
+        """
+        super().__init__(*args, **kwargs)
+        self.channel_axis = channel_axis
+        self.model_name = model_name
+        self._default_channel_axis = 2
+        if pool_strategy not in ('mean', 'max', None):
+            raise NotImplementedError(f'unknown pool_strategy: {self.pool_strategy}')
+        self.pool_strategy = pool_strategy
+
+    def post_init(self):
+        super().post_init()
+        import torchvision.models.video as models
+        if self.pool_strategy is not None:
+            self.pool_fn = getattr(np, self.pool_strategy)
+        self.model = getattr(models, self.model_name)(pretrained=True).eval()
+        self.to_device(self.model)
+
+    def _get_features(self, x):
```

Comment:
I had never seen this, but I think this model wont work with another model than the pretrained. I think we may need to block the model name to the default one and state it clearly in the readme. It is strange that we rely on this pytorch model to have these layers.